src/main/java/ca/on/oicr/pineryreports/reports/impl/LanesBillingReport.java

```diff
@@ -0,0 +1,464 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.ParseException;
+
+import com.google.common.collect.Sets;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pineryreports.data.ColumnDefinition;
+import ca.on.oicr.pineryreports.reports.TableReport;
+import ca.on.oicr.pineryreports.util.CommonOptions;
+import ca.on.oicr.ws.dto.InstrumentDto;
+import ca.on.oicr.ws.dto.InstrumentModelDto;
+import ca.on.oicr.ws.dto.RunDto;
+import ca.on.oicr.ws.dto.RunDtoPosition;
+import ca.on.oicr.ws.dto.RunDtoSample;
+import ca.on.oicr.ws.dto.SampleDto;
+
+public class LanesBillingReport extends TableReport {
+  
+  private static class DetailedObject {
+    private final String instrumentModel;
+    private final String instrumentName;
+    private final String laneContents;
+    private final String laneNumber;
+    private final String project;
+    private final BigDecimal projLibsPercent; // percentage of libs in lane which belong to given project
+    private final String runEndDate;
+    private final String runName;
+    private final String runStatus;
+    
+    public DetailedObject (String runEndDate, String instrumentName, String instrumentModel, String runName, 
+        String laneNumber, String project, BigDecimal libsPercent, String status, String laneContents) {
+      this.runEndDate = timeStringToYyyyMmDd(runEndDate);
+      this.instrumentName = instrumentName;
+      this.instrumentModel = instrumentModel;
+      this.runName = runName;
+      this.laneNumber = laneNumber;
+      this.project = project;
+      this.projLibsPercent = libsPercent;
+      this.runStatus = status;
+      this.laneContents = laneContents;
+    }
+
+    public String getInstrumentModel() {
+      return instrumentModel;
+    }
+    
+    public String getInstrumentName() {
+      return instrumentName;
+    }
+    
+    public String getLaneContents() {
+      return laneContents;
+    }
+
+    public String getLaneNumber() {
+      return laneNumber;
+    }
+
+    public String getProject() {
+      return project;
+    }
+    
+    public String getPrintableProjLibsPercent() {
+      return projLibsPercent.toPlainString();
+    }
+    
+    public BigDecimal getProjLibsPercent() {
+      return projLibsPercent;
+    }
+
+    public String getRunEndDate() {
+      return runEndDate;
+    }
+
+    public String getRunName() {
+      return runName;
+    }
+    
+    public String getRunStatus() {
+      return runStatus;
+    }
+    
+    private static final Comparator<DetailedObject> detailedComparator = new Comparator<DetailedObject>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)